### PR TITLE
Fix: Incorrect location for padded-blocks (fixes #4913)

### LINKED ISSUE #4913: Bug: Incorrect location for padded-blocks
Using master, the location of the error for closing padded block is wrong:
```
{
  
  a();
}
```
produces:
```
4:2 - Block must be padded by blank lines. (padded-blocks)
```
Column number should be `1`

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/29755559-bug-incorrect-location-for-padded-blocks?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by alberto**
Issue hasn't been yet marked as accepted, but maybe it's easier to see the code in the PR.

### CODE REVIEW

File: lib/rules/padded-blocks.js [hunk 1]
```diff
@@ -96,9 +96,13 @@ module.exports = function(context) {
                 if (!blockHasTopPadding) {
                     context.report(node, ALWAYS_MESSAGE);
                 }
-
                 if (!blockHasBottomPadding) {
-                    context.report(node, node.loc.end, ALWAYS_MESSAGE);
+                    console.log(node);
```
Comment (nzakas) [inline lib/rules/padded-blocks.js]: Oops?

Comment (alberto) [inline lib/rules/padded-blocks.js]: Ouch!

**Comment by nzakas** [comment]
LGTM

File: lib/rules/padded-blocks.js [hunk 2]
```diff
@@ -96,9 +96,12 @@ module.exports = function(context) {
                 if (!blockHasTopPadding) {
                     context.report(node, ALWAYS_MESSAGE);
                 }
-
                 if (!blockHasBottomPadding) {
-                    context.report(node, node.loc.end, ALWAYS_MESSAGE);
+                    context.report({
+                        node: node,
+                        loc: {line: node.loc.end.line, column: node.loc.end.column - 1 },
```
Comment (michaelficarra) [inline lib/rules/padded-blocks.js]: Doesn't this just mean there's an off-by-one in the parser? Why would we have to do this offset?

Comment (alberto) [inline lib/rules/padded-blocks.js]: @michaelficarra both ranges and locs are end-exclusive (in math notation `[start,end)`) I believe that is intentional. can someone from @eslint/eslint-team @eslint/espree-team confirm?

**Pull request merged**